src/hokusai-app-updater/examples/add_timeout_to_exec_probe.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python
+
+# run this script from the root of a local repo of a hokusai-managed project
+#
+# edit a project's staging/production hokusai specs
+# specifically on Kubernetes Deployment resources
+# specifically their livenessProbe and readinessProbe configs
+# specifically those that use 'exec' as probing method
+# add timeoutSeconds to those probes if they don't have the setting
+#
+# for example, given the below livenessProbe that uses exec
+# and a DESIRED_TIMEOUT_SECONDS of 3 seconds:
+#
+#        livenessProbe:
+#          exec:
+#            command:
+#            - pgrep
+#            - -f
+#            - sidekiq
+#          initialDelaySeconds: 30
+#          periodSeconds: 30
+#
+# produce this:
+#
+#        livenessProbe:
+#          exec:
+#            command:
+#            - pgrep
+#            - -f
+#            - sidekiq
+#          initialDelaySeconds: 30
+#          periodSeconds: 30
+#          timeoutSeconds: 3
+
+import glob
+import os
+import sys
+import yaml
+
+def probe(probe_type, container_spec):
+  ''' add missing timeoutSeconds to any existing probe type that uses exec '''
+  if probe_type in container_spec and 'exec' in container_spec[probe_type] and 'timeoutSeconds' not in container_spec[probe_type]:
+        container_spec[probe_type]['timeoutSeconds'] = DESIRED_TIMEOUT_SECONDS
+
+# adjust this to preference
+DESIRED_TIMEOUT_SECONDS = 3
+
+# all types  of probes in a Deployment spec
+PROBE_TYPES = ['livenessProbe', 'readinessProbe']
+
+base = 'hokusai'
+
+# glob all staging.yml..., production.yml..., which might not exist
+staging_files = glob.glob(base + '/staging.yml*')
+prod_files = glob.glob(base + '/production.yml*')
+
+tmpfile = base + '/tmp.yml'
+
+for files in staging_files, prod_files:
+  if files:
+    # go with the first file
+    path = files[0]
+  else:
+    # no files found, move on
+    continue
+
+  # figure env
+  if 'staging' in path:
+    env = 'staging'
+  else:
+    env = 'production'
+
+  # read in spec content
+  with open(path, 'r') as reader:
+    spec_content = reader.read()
+
+  # remove jinja syntax, otherwise yaml load fails
+  spec_content = spec_content.replace('{{ ', 'doublecurlyspace')
+  spec_content = spec_content.replace(' }}', 'spacedoublecurly')
```

Comment:
I struggled with this too... I'm sort of surprised it works globally, but if so great.

Reply:
didn't have a problem this time, but it doesn't cover [Gemini](https://github.com/artsy/gemini/blob/1b5292e86e2308ce888b79369714dcac1af8c623/hokusai/staging.yml#L49) which as `{% %}`. I tried adding:
```
spec_content = spec_content.replace('{% ', 'curlypercentspace')
spec_content = spec_content.replace(' %}', 'spacepercentcurly')
```
but the resulting YAML was still invalid, so I did Gemini manually.